cmd/controller-manager/app/options/options.go

```diff
@@ -34,39 +33,16 @@ import (
 	genericcontrollermanager "k8s.io/kubernetes/cmd/controller-manager/app"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/apis/componentconfig"
-	componentconfigv1alpha1 "k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
 )
 
 // GenericControllerManagerOptions is the common structure for a controller manager. It works with NewGenericControllerManagerOptions
 // and AddDefaultControllerFlags to create the common components of kube-controller-manager and cloud-controller-manager.
 type GenericControllerManagerOptions struct {
-	CloudProvider    *CloudProviderOptions
-	Debugging        *DebuggingOptions
-	GenericComponent *GenericComponentConfigOptions
-	KubeCloudShared  *KubeCloudSharedOptions
-
-	AttachDetachController           *AttachDetachControllerOptions
-	CSRSigningController             *CSRSigningControllerOptions
-	DaemonSetController              *DaemonSetControllerOptions
-	DeploymentController             *DeploymentControllerOptions
-	DeprecatedFlags                  *DeprecatedControllerOptions
-	EndPointController               *EndPointControllerOptions
-	GarbageCollectorController       *GarbageCollectorControllerOptions
-	HPAController                    *HPAControllerOptions
-	JobController                    *JobControllerOptions
-	NamespaceController              *NamespaceControllerOptions
-	NodeIpamController               *NodeIpamControllerOptions
-	NodeLifecycleController          *NodeLifecycleControllerOptions
-	PersistentVolumeBinderController *PersistentVolumeBinderControllerOptions
-	PodGCController                  *PodGCControllerOptions
-	ReplicaSetController             *ReplicaSetControllerOptions
-	ReplicationController            *ReplicationControllerOptions
-	ResourceQuotaController          *ResourceQuotaControllerOptions
-	SAController                     *SAControllerOptions
-	ServiceController                *ServiceControllerOptions
-
-	Controllers               []string
-	ExternalCloudVolumePlugin string
+	CloudProvider     *CloudProviderOptions
```

Comment:
actually I would get rid of `GenericControllerManagerOptions` and just put those 5 sub-options into the cloud and kube controller manager option. We don't have to over-abstract this generic part.

Reply:
sgtm